DOCSP-35976: Delete One usage example (#2821)
jenssegers/laravel-mongodb: docs/usage-examples/deleteOne.txt

.. _laravel-delete-one-usage:

=================
Delete a Document
=================

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: delete one, remove, code example

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

You can delete a document in a collection by retrieving a single document and calling
the ``delete()`` method on an Eloquent model or a query builder.

> Pass a query filter to the ``where()`` method, sort the matching documents, and call the

Reviewer: Suggestion: Is "query filter" defined somewhere in these docs yet? If so, it might be good to link to that somewhere. (ie: "For more information about query filters, see ... ")

> Pass a query filter to the ``where()`` method, sort the matching documents, and call the
> ``first()`` method to retrieve only the first document. Then, delete this matching document
> by calling the ``delete()`` method.

Reviewer: I think this paragraph could be more clear if it was presented as a list: 

```suggestion
Perform the following steps to delete a document:

1. Pass a query filter to the ``where()`` method
2. Sort the matching documents
3. Call the ``first()`` method to retrieve only the first document 
4. Delete this matching document by calling the ``delete()`` method.
```

Although, looking at the update usage example page, it looks like that might not be inline with the formatting being used in these docs. An alternative suggestion would be:

```suggestion
To delete a document, pass a query filter to the ``where()`` method, sort the matching documents, and call the
``first()`` method to retrieve only the first document. Then, delete this matching document
by calling the ``delete()`` method.
```

Author: Took your second suggestion!

Example
-------

This usage example performs the following actions:

> - Uses the ``Movie`` Eloquent model to represent the ``movies`` collection in the
>   ``sample_mflix`` database.
> - Deletes a document from the ``movies`` collection that matches a query filter.

Reviewer: I don't think we should have a period on these since they're not full sentences (subject is in the intro)

```suggestion
- Uses the ``Movie`` Eloquent model to represent the ``movies`` collection in the
  ``sample_mflix`` database.
- Deletes a document from the ``movies`` collection that matches a query filter.
```

Reviewer: Feel free to double-check me on this one - I'm not 100% sure if we still consider it a fragment with the subject implied in the intro

Author: Agreed - I think these are all fragments. Fixed!

.. _laravel-delete-one-usage:

=================
Delete a Document
=================

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: delete one, remove, code example

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

> You can delete a document in a collection by retrieving a single document and calling
> the ``delete()`` method on an Eloquent model or a query builder.

Reviewer 2: I don't understand the "or a query builder" at the end of this sentence.

```suggestion
You can delete a document in a collection by calling the ``delete()`` method on an Eloquent model that have been retrieved, or directly on a query builder.
```
